src/java/org/apache/cassandra/utils/LazyCloseableRef.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils;
+
+import java.util.function.Supplier;
+import javax.annotation.concurrent.NotThreadSafe;
+
+public abstract class LazyCloseableRef<T extends AutoCloseable> implements Closeable, Supplier<T>
```

Comment:
Why do we need this to be closeable? The builders don't manage these resources.

I think changing the getters to `open...` as described above will add clarity and make it less likely for incorrect resource management to be introduced later.

Reply:
thanks for the idea, will try and see how it looks